win32ss/user/user32/misc/exticon.c

```diff
@@ -814,9 +848,16 @@ UINT WINAPI PrivateExtractIconExW (
 	TRACE("%s %d %p %p %d\n",
 	debugstr_w(lpwstrFile),nIndex,phIconLarge, phIconSmall, nIcons);
 
-	if (nIndex == -1)
-	  /* get the number of icons */
-	  return ICO_ExtractIconExW(lpwstrFile, NULL, 0, 0, 0, 0, NULL, LR_DEFAULTCOLOR);
+#ifdef __REACTOS__
+    if ((nIndex == -1) || (!phIconSmall && !phIconLarge))
+      /* get the number of icons */
```

Comment:
nitpick: I would move that comment to the line of the if.

Reply:
better not to be changed for the sake of symmetry to the non-ifdef-REACTOS-part below.